packages/compiler/core/checker.ts

```diff
@@ -3532,7 +3532,16 @@ export function createChecker(program: Program): Checker {
         kind: "Function",
         call(...args: Type[]): Type {
           const retval = ref.value!(program, ...marshalProjectionArguments(args));
-          return marshalProjectionReturn(retval);
+          try {
+            return marshalProjectionReturn(retval);
+          } catch (e) {
+            if (e instanceof ProjectionError) {
+              throw new Error(
+                `Can't marshal value "${retval}" returned from JS function "${node.sv}" into cadl`
+              );
```

Comment:
Is the message in `e` not of any use here? Do we need to change the type from ProjectionError to error?

Reply:
That's the error returned by marshasl

```
    throw new ProjectionError("Can't marshal value returned from JS function into cadl");
```

but yes I guess could change to projection error

Reply:
Either way sounds fine.